diff --git a/drivers/net/wireless/Makefile b/drivers/net/wireless/Makefile
index a61cf6c90..b4da9cc8a 100644
--- a/drivers/net/wireless/Makefile
+++ b/drivers/net/wireless/Makefile
@@ -1,3 +1,4 @@
+obj-m += mediatek/
 # SPDX-License-Identifier: GPL-2.0
 #
 # Makefile for the Linux Wireless network device drivers.
diff --git a/drivers/net/wireless/mediatek/Makefile b/drivers/net/wireless/mediatek/Makefile
index 806172659..f1410d008 100644
--- a/drivers/net/wireless/mediatek/Makefile
+++ b/drivers/net/wireless/mediatek/Makefile
@@ -1,3 +1,4 @@
+obj-m += mt76/
 # SPDX-License-Identifier: GPL-2.0-only
 obj-$(CONFIG_MT7601U)	+= mt7601u/
 obj-$(CONFIG_MT76_CORE)	+= mt76/
diff --git a/drivers/net/wireless/mediatek/mt76/Makefile b/drivers/net/wireless/mediatek/mt76/Makefile
index c78ae4b89..539ebfe1c 100644
--- a/drivers/net/wireless/mediatek/mt76/Makefile
+++ b/drivers/net/wireless/mediatek/mt76/Makefile
@@ -1,3 +1,8 @@
+CONFIG_MT76_CORE := m
+CONFIG_MT76_USB := m
+CONFIG_MT76x02_LIB := m
+CONFIG_MT76x02_USB := m
+obj-m += mt76x2/
 # SPDX-License-Identifier: GPL-2.0-only
 obj-$(CONFIG_MT76_CORE) += mt76.o
 obj-$(CONFIG_MT76_USB) += mt76-usb.o
diff --git a/drivers/net/wireless/mediatek/mt76/mt76x2/Makefile b/drivers/net/wireless/mediatek/mt76/mt76x2/Makefile
index caf089538..032fcd7fa 100644
--- a/drivers/net/wireless/mediatek/mt76/mt76x2/Makefile
+++ b/drivers/net/wireless/mediatek/mt76/mt76x2/Makefile
@@ -1,3 +1,5 @@
+CONFIG_MT76x2_COMMON := m
+CONFIG_MT76x2U := m
 # SPDX-License-Identifier: GPL-2.0-only
 obj-$(CONFIG_MT76x2_COMMON) += mt76x2-common.o
 obj-$(CONFIG_MT76x2E) += mt76x2e.o
diff --git a/net/Makefile b/net/Makefile
index 103cd8d61..50060dce8 100644
--- a/net/Makefile
+++ b/net/Makefile
@@ -1,3 +1,5 @@
+obj-m += wireless/
+obj-m += mac80211/
 # SPDX-License-Identifier: GPL-2.0
 #
 # Makefile for the linux networking.
diff --git a/net/mac80211/Makefile b/net/mac80211/Makefile
index b8de44da1..d6956b92a 100644
--- a/net/mac80211/Makefile
+++ b/net/mac80211/Makefile
@@ -1,3 +1,4 @@
+CONFIG_MAC80211 := m
 # SPDX-License-Identifier: GPL-2.0
 obj-$(CONFIG_MAC80211) += mac80211.o
 
diff --git a/net/wireless/Makefile b/net/wireless/Makefile
index 527ae669f..6983a2ff9 100644
--- a/net/wireless/Makefile
+++ b/net/wireless/Makefile
@@ -1,3 +1,4 @@
+CONFIG_CFG80211 := m
 # SPDX-License-Identifier: GPL-2.0
 obj-$(CONFIG_CFG80211) += cfg80211.o
 obj-$(CONFIG_LIB80211) += lib80211.o
