diff --git a/modules.bzl b/modules.bzl
index 52d92225a..e0002ab33 100644
--- a/modules.bzl
+++ b/modules.bzl
@@ -39,7 +39,6 @@ _COMMON_GKI_MODULES_LIST = [
     "drivers/usb/serial/ftdi_sio.ko",
     "drivers/usb/serial/usbserial.ko",
     "kernel/kheaders.ko",
-    "lib/crypto/libarc4.ko",
     "mm/zsmalloc.ko",
     "net/6lowpan/6lowpan.ko",
     "net/6lowpan/nhc_dest.ko",
@@ -64,7 +63,6 @@ _COMMON_GKI_MODULES_LIST = [
     "net/l2tp/l2tp_ppp.ko",
     "net/mac802154/mac802154.ko",
     "net/nfc/nfc.ko",
-    "net/rfkill/rfkill.ko",
     "net/tipc/diag.ko",
     "net/tipc/tipc.ko",
 ]
