diff --git a/fs/Kconfig b/fs/Kconfig
index 410a8f553..7e212ecc3 100644
--- a/fs/Kconfig
+++ b/fs/Kconfig
@@ -387,4 +387,5 @@ source "fs/unicode/Kconfig"
 config IO_WQ
 	bool
 
+source "fs/nomount/Kconfig"
 endmenu
diff --git a/fs/Makefile b/fs/Makefile
index 102e4cd2b..3e1928266 100644
--- a/fs/Makefile
+++ b/fs/Makefile
@@ -135,3 +135,5 @@ obj-$(CONFIG_EFIVAR_FS)		+= efivarfs/
 obj-$(CONFIG_EROFS_FS)		+= erofs/
 obj-$(CONFIG_VBOXSF_FS)		+= vboxsf/
 obj-$(CONFIG_ZONEFS_FS)		+= zonefs/
+
+obj-$(CONFIG_NOMOUNT) += nomount/
