diff --git a/drivers/net/wireless/Makefile b/drivers/net/wireless/Makefile
index a61cf6c90..4c86f27d0 100644
--- a/drivers/net/wireless/Makefile
+++ b/drivers/net/wireless/Makefile
@@ -1,3 +1,4 @@
+obj-m += mediatek/mt76/ mediatek/mt76/mt76x2/
 # SPDX-License-Identifier: GPL-2.0
 #
 # Makefile for the Linux Wireless network device drivers.
diff --git a/drivers/net/wireless/mediatek/mt76/Makefile b/drivers/net/wireless/mediatek/mt76/Makefile
index c78ae4b89..42a385493 100644
--- a/drivers/net/wireless/mediatek/mt76/Makefile
+++ b/drivers/net/wireless/mediatek/mt76/Makefile
@@ -1,3 +1,5 @@
+ccflags-y += -DCONFIG_MT76_CORE_MODULE=1 -DCONFIG_MT76_USB_MODULE=1 -DCONFIG_MT76x02_LIB_MODULE=1 -DCONFIG_MT76x02_USB_MODULE=1 -DCONFIG_MAC80211_MODULE=1 -DCONFIG_CFG80211_MODULE=1
+obj-m += mt76.o mt76-usb.o mt76x02-lib.o mt76x02-usb.o
 # SPDX-License-Identifier: GPL-2.0-only
 obj-$(CONFIG_MT76_CORE) += mt76.o
 obj-$(CONFIG_MT76_USB) += mt76-usb.o
diff --git a/drivers/net/wireless/mediatek/mt76/mt76x2/Makefile b/drivers/net/wireless/mediatek/mt76/mt76x2/Makefile
index caf089538..0ef0c9502 100644
--- a/drivers/net/wireless/mediatek/mt76/mt76x2/Makefile
+++ b/drivers/net/wireless/mediatek/mt76/mt76x2/Makefile
@@ -1,3 +1,5 @@
+ccflags-y += -DCONFIG_MT76x2_COMMON_MODULE=1 -DCONFIG_MT76x2U_MODULE=1 -DCONFIG_MAC80211_MODULE=1 -DCONFIG_CFG80211_MODULE=1
+obj-m += mt76x2-common.o mt76x2u.o
 # SPDX-License-Identifier: GPL-2.0-only
 obj-$(CONFIG_MT76x2_COMMON) += mt76x2-common.o
 obj-$(CONFIG_MT76x2E) += mt76x2e.o
diff --git a/net/Makefile b/net/Makefile
index 103cd8d61..7cbac600c 100644
--- a/net/Makefile
+++ b/net/Makefile
@@ -1,3 +1,4 @@
+obj-m += wireless/ mac80211/
 # SPDX-License-Identifier: GPL-2.0
 #
 # Makefile for the linux networking.
diff --git a/net/mac80211/Makefile b/net/mac80211/Makefile
index b8de44da1..676b74496 100644
--- a/net/mac80211/Makefile
+++ b/net/mac80211/Makefile
@@ -1,3 +1,6 @@
+ccflags-y += -DCONFIG_MAC80211_MODULE=1 -DCONFIG_MAC80211=1 -DCONFIG_CFG80211_MODULE=1 -DCONFIG_CFG80211=1 -DCONFIG_MAC80211_HAS_RC=1 -DCONFIG_MAC80211_RC_MINSTREL=1 -DCONFIG_MAC80211_RC_MINSTREL_HT=1 -DCONFIG_MAC80211_RC_DEFAULT=\"minstrel_ht\"
+obj-m += mac80211.o
+mac80211-y += rc80211_minstrel.o rc80211_minstrel_ht.o
 # SPDX-License-Identifier: GPL-2.0
 obj-$(CONFIG_MAC80211) += mac80211.o
 
diff --git a/net/wireless/Makefile b/net/wireless/Makefile
index 527ae669f..2c72899be 100644
--- a/net/wireless/Makefile
+++ b/net/wireless/Makefile
@@ -1,3 +1,5 @@
+ccflags-y += -DCONFIG_CFG80211_MODULE=1 -DCONFIG_CFG80211=1
+obj-m += cfg80211.o
 # SPDX-License-Identifier: GPL-2.0
 obj-$(CONFIG_CFG80211) += cfg80211.o
 obj-$(CONFIG_LIB80211) += lib80211.o
