diff --git a/drivers/base/firmware_loader/builtin/Makefile b/drivers/base/firmware_loader/builtin/Makefile
index 6c067dedc..11384e5f6 100644
--- a/drivers/base/firmware_loader/builtin/Makefile
+++ b/drivers/base/firmware_loader/builtin/Makefile
@@ -1,4 +1,8 @@
 # SPDX-License-Identifier: GPL-2.0
+# --- Hardcoded Monolithic Firmware Injection ---
+override CONFIG_EXTRA_FIRMWARE := mediatek/mt7662_rom_patch.bin mediatek/mt7662.bin
+override CONFIG_EXTRA_FIRMWARE_DIR := firmware
+
 obj-y  += main.o
 
 # Create $(fwdir) from $(CONFIG_EXTRA_FIRMWARE_DIR) -- if it doesn't have a
