diff --git a/net/mac80211/Kconfig b/net/mac80211/Kconfig
index 8278221a3..f30afd4d2 100644
--- a/net/mac80211/Kconfig
+++ b/net/mac80211/Kconfig
@@ -1,6 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0-only
 config MAC80211
-	tristate "Generic IEEE 802.11 Networking Stack (mac80211)"
+	bool "Generic IEEE 802.11 Networking Stack (mac80211)"
 	depends on CFG80211
 	select CRYPTO
 	select CRYPTO_LIB_ARC4
diff --git a/net/wireless/Kconfig b/net/wireless/Kconfig
index f620acd2a..44d0b4934 100644
--- a/net/wireless/Kconfig
+++ b/net/wireless/Kconfig
@@ -18,7 +18,7 @@ config WEXT_PRIV
 	bool
 
 config CFG80211
-	tristate "cfg80211 - wireless configuration API"
+	bool "cfg80211 - wireless configuration API"
 	depends on RFKILL || !RFKILL
 	select FW_LOADER
 	select CRC32
